icu4c/source/test/intltest/numbertest.h

```diff
@@ -114,16 +114,38 @@ class NumberFormatterApiTest : public IntlTestWithFieldPosition {
     DecimalFormatSymbols SWISS_SYMBOLS;
     DecimalFormatSymbols MYANMAR_SYMBOLS;
 
-    void assertFormatDescending(const char16_t* message, const char16_t* skeleton,
-                                const UnlocalizedNumberFormatter& f, Locale locale, ...);
+    /**
+     * skeleton is the full length skeleton, which must round-trip.
+     *
+     * conciseSkeleton should be the shortest available skeleton.
+     * The concise skeleton can be read but not printed.
+     */
+    void assertFormatDescending(
+      const char16_t* message,
```

Comment:
What is the reason/benefit of using char16_t* here instead of StringPiece?  Or are they not designed to be compatible?

Reply:
StringPiece is an 8-bit char string.  UnicodeString is a 16-bit char16_t string.  I could make the argument a UnicodeString, but the style of implicitly converting a raw string to a UnicodeString is somewhat frowned upon.  Generally I like to pass raw strings until I actually need to do work on them.